ipld/get_shares.go

```diff
@@ -76,13 +75,15 @@ func GetShares(ctx context.Context, bGetter blockservice.BlockGetter, root cid.C
 			// processing of each other
 			pool.Submit(func() {
 				ctx, span := tracer.Start(ctx, "process-job")
-				defer span.End()
 				defer wg.Done()
+				defer span.End()
```

Comment:
@renaynay investigated this and found that multiple defers are LIFO.

Reply:
yes and why this order is needed here, don't really get it?

Reply:
Then you won't need to record error with attributes on loc 170 as well

Reply:
Should the span always be the last thing to be closed? Felt/sounded right at first to end the span, then "close" the job, but I can see how that probably doesn't make sense, since the span should encompass ("span") over absolutely everything

Reply:
Yes